kbase-extension/static/kbase/js/util/icon.js

```diff
@@ -1,150 +1,246 @@
+        for (let i=num_stacked_circles; i > 0; i--) {
+            let stacked_color = color
+                ? 'rgb(' + cmax(r + i * cstep)  + ',' +
+                    cmax(g + i * cstep) + ',' + cmax(b + i * cstep) + ')'
+                : '';
+            let style = color ? { color: stacked_color } : {};
+            layers.push(
+                span({
+                    class: `${cssBaseName}__stack--l${i} fa fa-stack-2x fa-${shape}`,
+                    style: style,
+                }),
+                // white separator
+                span({
+                    class: `${cssBaseName}__outline--l${i} fa fa-stack-2x fa-${shape}`,
+                })
+            );
+        }
+        return layers;
+    }
+
+    /**
+     * @method _makeIcon
+     * @private
+     * Creates an HTML string that renders an icon
+     * @param {string} iconType  the type of icon to produce,
+     *                              e.g. 'app', 'app-toolbar', 'data', 'generic', etc.
+     * @param {object} iconClass  the class of the icon, e.g. 'fa-cube'
+     * @param {string} shape      shape for the icon background;
+     *                              typically 'square' or 'circle'
+     * @param {string} color      colour for the icon background (hex or RGB)
+     * @param {bool}   stacked    whether or not the icon should be stacked
+     */
+
+    function _makeIcon(iconType, iconClass, shape = 'square', color = null, stacked = false) {
+
+        const style = color ? {color: color} : {};
+        const layers = stacked ? _makeStackLayers(shape, color) : [];
+        iconClass = iconClass.replace('icon ', '');
+
+        return span({
+            class: `${cssBaseName}__container--${iconType} fa-stack`,
+        }, layers.concat(
+            span({
+                class: `${cssBaseName}__icon_background--${iconType} fa fa-${shape} fa-stack-2x`,
+                style: style,
+            }),
+            span({
+                class: `${cssBaseName}__icon--${iconType} fa fa-inverse fa-stack-1x ${iconClass}`,
+            })
+        ));
+    }
+
 
     /**
      * Legacy code that overwrites any existing DOM elements in params.elt. Used for passing in
-     * a set of parameters to buildDataIcon as an object instead of a list.
+     * a set of parameters to makeDataIcon as an object instead of a list.
      * @param {object} params - The set of parameters to use.
      * @param {object} params.elt - The JQuery element containing the logo to overwrite.
      * @param {string} params.type - The type of object to build a logo from.
      * @param {boolean} params.stacked - If true, creates a stacked effect of multiple icons on each other
-     * @param {int} params.indent - How many spaces to indent (can be null or undefined)
      */
     function overwriteDataIcon(params) {
-        return buildDataIcon(params.elt, params.type, params.stacked, params.indent);
+        return makeDataIcon(params.elt, params.type, params.stacked);
     }
```

Comment:
Couldn't find anywhere that `overwriteDataIcon` was used -- is there any point in keeping it?